Docs: mention that IDs for items and locations can overlap (#2854)
ArchipelagoMW/Archipelago: docs/world api.md

Each location has a `name` and an `address` (hereafter referred to as an `id`), is placed in a Region, has access rules,
and has a classification. The name needs to be unique within each game and must not be numeric (must contain least 1
letter or symbol). The ID needs to be unique across all games, and is best kept in the same range as the item IDs.

> Locations and items can share IDs, so typically locations and items start at the same ID for a game.

Reviewer: Some other wordings were suggested on Discord. I think this one is still my preference:
```suggestion
Locations and items can share IDs, so typically a game's locations and items start at the same ID.
```

> Locations and items can share IDs, so typically locations and items start at the same ID for a game.

Reviewer 2: ```suggestion
Typically, a game's location and item IDs share a range and start at the same value.
```
